omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -257,6 +283,171 @@ else if(identifierSourceType.equals(Identifier_Pool)){
+        Object operation = updateBody.get("operation");
+        Object generateIdentifiers = updateBody.get("generateIdentifiers");
+        IdentifierSource identifierSourceToUpdate = Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(uuid);
+        if (reservedIdentifiersToUpload != null) {
+            List<String> reservedIdentifiersList = new ArrayList<String>(Arrays.asList(reservedIdentifiersToUpload.toString().split(",")));
+            if(identifierSourceToUpdate != null){
+               for (int counter = 0; counter < reservedIdentifiersList.size(); counter++) {
+                    if (StringUtils.isNotBlank(reservedIdentifiersList.get(counter))) {
+                        identifierSourceToUpdate.addReservedIdentifier(reservedIdentifiersList.get(counter));
+                    }
+                }
+                Context.getService(IdentifierSourceService.class).saveIdentifierSource(identifierSourceToUpdate);
+            }
+        }
+        if (operation != null && operation.toString().equals("uploadFromSource")) {
+            if(identifierSourceToUpdate != null){
+                IdentifierPool pool = (IdentifierPool) identifierSourceToUpdate;
+                Object batchSize = updateBody.get("batchSize");
+                if(pool != null && batchSize != null && StringUtils.isNumeric(batchSize.toString())){
+                    Context.getService(IdentifierSourceService.class).addIdentifiersToPool(pool, Integer.parseInt(batchSize.toString()));
+                }
+            }
+        }
+        if (operation != null&& operation.toString().equals("uploadFromFile")) {
+            Object identifiers = updateBody.get("identifiers");
+            if(identifierSourceToUpdate != null && identifiers != null){
+                List<String> ids = new ArrayList<String>(Arrays.asList(identifiers.toString().split(",")));
+                IdentifierPool pool = (IdentifierPool) identifierSourceToUpdate;
+                if(pool != null && ids != null){
+                    Context.getService(IdentifierSourceService.class).addIdentifiersToPool(pool, ids);
+                } 
+            }
+        }
+        if (generateIdentifiers != null) {
+            Object comment = updateBody.get("comment");
+            Object numberToGenerate = updateBody.get("numberToGenerate");
+            if (comment == null) {
+                comment = "Batch Export of " + numberToGenerate + " to file";
+            }
+            if(numberToGenerate == null){
+                numberToGenerate = 0;
+            }
+            if(identifierSourceToUpdate != null){
+                SimpleObject identifiers = new SimpleObject();
+                identifiers.add("identifiers", Context.getService(IdentifierSourceService.class).
+                        generateIdentifiers(identifierSourceToUpdate, Integer.parseInt(numberToGenerate.toString()), comment.toString()));
+                return identifiers;
```

Comment:
Why return these?

Reply:
@dkayiwa This section of code is handling the export of identifiers to the user. We have to return the identifiers. The user provides a number of identifiers to export and a comment and a comment. Identifiers are then generated and exported.

Reply:
Is that via an update REST call?

Reply:
@dkayiwa I wanted to put the code in the `doSearch()` method but because we are updating the `comment` fiedl i thought it would be better to put the code here.

Reply:
Isn't updating about modifying an existing?

Reply:
@dkayiwa True.

Reply:
@dkayiwa Will move the code to search.

Reply:
@dkayiwa I have a new issue arising. I need to convert the identifiers generated to `Ref Representation`. I tried to create a SimpleObject but that did not work. I have pushed the code.